docs/source/developers/python.rst

```diff
@@ -398,7 +398,7 @@ Now, build pyarrow:
 
    $ pushd arrow/python
    $ export PYARROW_PARALLEL=4
-   $ python setup.py build_ext --inplace
+   $ python -m pip install -e .
```

Comment:
```suggestion
   $ python -m pip install -e . --no-build-isolation --no-deps -v
```

Those additional flags make it more similar to what we did before. It's probably debatable whether those should be added (disabling build isolation and installing deps), but since in this tutorial we have set up all build dependencies in advance, not doing that gives unnecessary pip overhead. 
(this is also the command that I use locally, although for quick rebuilds I still use `python setup.py build_ext --inplace`)